### ISSUE TITLE
FrodoKEM-AES terribly slow without AESNI

### ISSUE THREAD

**Issue by randombit** [randombit/botan#4173]
I noticed that `frodo_kat_tests` takes forever (several minutes) in qemu builds, and that AES seemed worst affected. This replicates on a desktop machine.

With AES-NI

```
FrodoKEM-640-AES 832 KEM encrypt/sec; 1.20 ms/op 4812154 cycles/op (413 ops in 496 ms)
FrodoKEM-640-AES 825 KEM decrypt/sec; 1.21 ms/op 4852335 cycles/op (413 ops in 500 ms)
```

Using ``--clear-cpuid=sse2`` (to disable both AES-NI and vperm AES)

```
FrodoKEM-640-AES 57 KEM encrypt/sec; 17.25 ms/op 69152136 cycles/op (29 ops in 500 ms)
FrodoKEM-640-AES 58 KEM decrypt/sec; 17.23 ms/op 69049816 cycles/op (29 ops in 500 ms)
```

The problem seems to be that in https://github.com/randombit/botan/blob/master/src/lib/pubkey/frodokem/frodokem_aes/frodo_aes_generator.h the lamba only ever encrypts 1 block at a time. This is maximally pessimal. On AES-NI and co, and in vperm, it inhibits pipelining. And the bytesliced fallback always encrypts 2 blocks at a time...

Improving this would both alleviate the horrible performance on CPUs without AES hardware and also likely speed up Frodo on systems with AES support.

Indeed when we encrypt all of the blocks in parallel:

```
FrodoKEM-640-AES 1666 KEM encrypt/sec; 0.60 ms/op 2405453 cycles/op (813 ops in 488 ms)
FrodoKEM-640-AES 1625 KEM decrypt/sec; 0.62 ms/op 2465076 cycles/op (813 ops in 500 ms)
```

or without AES-NI

```
FrodoKEM-640-AES 131 KEM encrypt/sec; 7.63 ms/op 30575435 cycles/op (66 ops in 503 ms)
FrodoKEM-640-AES 130 KEM decrypt/sec; 7.65 ms/op 30666740 cycles/op (66 ops in 505 ms)
```

So basically a 2x speedup for either case.
